src/com/talar/server/Server.java

```diff
@@ -0,0 +1,258 @@
+package com.talar.server;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.InetAddress;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.text.SimpleDateFormat;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Date;
+
+public class Server {
+
+        private int port = 0;
+        private ServerSocket server = null;
+        private BufferedInputStream bis = null;
+        private BufferedOutputStream bos = null;
+
+        public Server() {
+
+        }
+
+        public void start() {
+                Socket socket = null;
+                boolean loop = false;
+                
+                try {
+                        server = new ServerSocket(port);
+                        loop = true;
+                        
+                        while(loop) {
+                                System.out.println("Server started on port " + getPort());
+
+                                socket = server.accept();
+                                
+                                InetAddress addr = socket.getInetAddress();
+                                String ipAddr = addr.getHostAddress();
+                                System.out.println("Client connected " + ipAddr + ":" + socket.getLocalPort());
+                                
+                                boolean incoming = true;
+                                while(incoming) {
+                                        InputStream in = socket.getInputStream();
+                                        OutputStream out = socket.getOutputStream();
+                                        bis = new BufferedInputStream(in);
+                                        bos = new BufferedOutputStream(out);
+        
+                                        StringBuilder sb = new StringBuilder();
+                                        
+                                        while(true) {
+                                                int c = bis.read();
+        
+                                                if(c == 0) {
+                                                        break;
+                                                }
+        
+                                                sb.append((char)c);
+                                        }
+        
+                                        String receivedMessage = sb.toString();
+                                        String outMessage = "Server Received: \0" + sb.toString();
```

Comment:
This is confusing calling it `outMessage` it is just printing what the server will send. Better keep it as a system.out message .. 1 line